stock_picking_invoice_product_group/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Alex Comba - Agile Business Group
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api, fields
+
+
+class StockInvoiceOnshipping(models.TransientModel):
+    _inherit = 'stock.invoice.onshipping'
+
+    group_type = fields.Selection(
+        [('group_by_product', 'Group by product'),
+         ('group_by_product_category', 'Group by product category')
+         ],
+        string='Group Type',
+    )
+
+    @api.model
+    def _group_moves_by_group_type(self, moves, group_type):
+        """Return moves grouped by group_type"""
+        grouped_moves = {}
+        for move in moves:
+            partner = move.picking_id.partner_id
+            product = move.product_id
+            categ = product.categ_id
+            if group_type == 'group_by_product':
+                key = (partner.id, product.id)
+            else:
```

Comment:
Here, in the else statement, is basically developed an implicit "default" for the field group_type: if group_type has not the value 'group_by_product', then the system should suppose that it has the value 'group_by_product_category'. This is actually not correct (the field is not declared as required, therefore it could already assume the None value) and not safe in the perspective of further developments.

I think that a better way to achieve the same result could be to explicitly set a default for the field 'group_type' and to declare it as required (either on the model definition or on the view definition).

In any case, I believe it's better to use an elif branch in order to handle the 'group_by_product_category' case, reserving the else statement to eventually handle any unexpected value.